src/v2i-hub/docs/Programming_Guide.md

```diff
@@ -0,0 +1,1233 @@
+# Programming Guide
+
+The V2X Hub Programming Guide was developed during the V2I Reference Implementation project for developers of plugins for the V2X Hub in supporting Connected Vehicle (CV) technology. This programming guide gives details on how to create a plugin for the V2X Hub based on the Example Plugin created by Battelle. The Example Plugin can be obtained from OSADP, along with the other V2X Hub software that you will need to compile and setup to do your own V2X Hub development.
```

Comment:
Maybe add a link to ` Example Plugin `? Not sure Who is Battelle, and OSADP?

Reply:
This comes directly from https://github.com/usdot-fhwa-OPS/V2X-Hub/blob/develop/docs/V2I_Hub_PluginProgrammingGuide_Final.pdf which can also be found as a word doc on confluence here https://usdot-carma.atlassian.net/wiki/spaces/V2XH/pages/1508311057/V2X+Hub+Project+Documentation?preview=/1508311057/1508311084/V2X_Hub_PluginProgrammingGuide_3-2.docx . I have simply copied it and reformatted it into markdown for easier maintenance. 

As for a link the the SampleBsmPlugin, I included it in the parent page. I can link this one to the ExamplePlugin but I really want to work to consolidate these examples and have a single one. I was hoping to include this programming guide now without major revision and start revising individual parts as we continue to develop.